numba/tests/test_dispatcher.py

```diff
@@ -1266,6 +1266,17 @@ def test_closure(self):
             self.assertPreciseEqual(f(3), 12) # 3 + 9 = 12
             self.check_pycache(5) # 1 nbi, 4 nbc
 
+    def test_first_class_function(self):
+        mod = self.import_module()
+        f = mod.first_class_function_usecase
+        assert f(mod.first_class_function_mul, 1) == 1
+        assert f(mod.first_class_function_mul, 10) == 100
+        assert f(mod.first_class_function_add, 1) == 2
+        assert f(mod.first_class_function_add, 10) == 20
```

Comment:
```suggestion
        self.assertEqual(f(mod.first_class_function_mul, 1), 1)
        self.assertEqual(f(mod.first_class_function_mul, 10), 100)
        self.assertEqual(f(mod.first_class_function_add, 1), 2)
        self.assertEqual(f(mod.first_class_function_add, 10), 20)
```
Numba uses the `unittest.*` assertions.